Commands/Commands/top10.js

```diff
@@ -0,0 +1,104 @@
+var Entities = require('html-entities').AllHtmlEntities
+var config = require('../../config.js')
+const entities = new Entities()
+const Dash = require('rethinkdbdash')
+const r = new Dash()
+
+var commands = []
+
+commands.initializeTop = {
+  internal: true,
+  fn: function (bot, uv) {
+    setInterval(() => generateTop(bot, uv), 3600000) // update the top 10 hourly
+  }
+}
+
+commands.regenerate = {
+  adminOnly: true,
+  modOnly: false,
+  fn: function (bot, msg, suffix, uv) {
+    msg.reply('regenerating the top 10, this could take a while...').then(() => {
+      generateTop(bot, uv)
+    })
+  }
+}
+
+function generateTop (bot, uv) {
+  return new Promise(function(resolve, reject) {
+    let channel = bot.Channels.find(c => c.name === 'top-10-suggestions')
+    let messages
+    channel.fetchMessages().then(msgs => {
+      messages = msgs.messages.filter(y => y.author.id === bot.User.id)
+    })
+    uv.v1.loginAsOwner().then(f => {
+      f.get(`forums/${config.uservoice.forumId}/suggestions.json`).then(data => {
+        for (let suggestion of data.suggestions) {
+          generateEmbed(suggestion).then(embed => {
+            let message = messages.pop()
+            if (!message) {
+              channel.sendMessage('', false, embed).then(msg => {
+                r.db('DFB').table('queue').insert({
+                  id: msg.id,
+                  type: 'upvoteOnly',
+                  UvId: suggestion.id
+                }).then(() => {
+                  msg.addReaction({
+                    id: '302138464986595339',

```

Comment:
I wonder if we could put the voting reaction id to a variable somewhere considering how much we use it.